### PR TITLE
Docs: Fix links broken by deprecation of docs on grafana repo

### PR DESCRIPTION
**Pull request by josmperez**
Fix links containing path grafana/latest/developers/plugins so that they point to updated location on developer portal.

### CODE REVIEW

File: docusaurus/docs/introduction/backend.md [hunk 1]
```diff
@@ -97,4 +97,4 @@ The Grafana HTTP API offers an endpoint (`/api/plugins/<plugin id>/metrics`) tha
 
 #### Streaming
 
-The streaming capability allows a backend plugin to handle data source queries that are streaming. For more information, refer to [Build a streaming data source plugin](https://grafana.com/docs/grafana/latest/developers/plugins/create-a-grafana-plugin/develop-a-plugin/build-a-streaming-data-source-plugin/).
\ No newline at end of file
+The streaming capability allows a backend plugin to handle data source queries that are streaming.
```
Comment (sympatheticmoose) [inline docusaurus/docs/introduction/backend.md:100]:
```suggestion
The streaming capability allows a backend plugin to handle data source queries that are streaming. 

For more information, refer to the [Datasource Steaming Backend](https://github.com/grafana/grafana-plugin-examples/tree/main/examples/datasource-streaming-backend-websocket) example. 
```

File: docusaurus/docs/migration-guides/update-from-grafana-versions/v9.3.x-9.4.x.md [hunk 2]
```diff
@@ -145,8 +145,8 @@ useEffect(() => {
...
-](https://grafana.com/docs/grafana/latest/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-user-header-for-the-logged-in-user) or [Forward user header for the logged-in user](https://grafana.com/docs/grafana/latest/developers/plugins/create-a-grafana-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-user-header-for-the-logged-in-user) for example usages.
+We recommended to use the `<request>.GetHTTPHeader` or `<request>.GetHTTPHeaders` methods when retrieving forwarded HTTP headers. See [Forward OAuth identity for the logged-in user](../../create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-oauth-identity-for-the-logged-in-user), [Forward cookies for the logged-in user
+](../../create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-user-header-for-the-logged-in-user) or [Forward user header for the logged-in user](../../create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-user-header-for-the-logged-in-user) for example usages.
```
Comment (sympatheticmoose) [inline docusaurus/docs/migration-guides/update-from-grafana-versions/v9.3.x-9.4.x.md:149]:
```suggestion
](../../create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-cookies-for-the-logged-in-user) or [Forward user header for the logged-in user](../../create-a-plugin/extend-a-plugin/add-authentication-for-data-source-plugins#forward-user-header-for-the-logged-in-user) for example usages.
```

The second link is wrong

File: docusaurus/docs/get-started/get-started.mdx [hunk 3]
```diff
@@ -180,7 +180,7 @@ App and data source plugins can have a backend component written in Go. Backend
 - Connect to non-HTTP services to which a browser normally can’t connect. For example, SQL database servers.
 - Keep state between users. For example, query caching for data sources.
 - Use custom authentication methods and/or authorization checks that aren’t supported in Grafana.
-- Use a custom data source request proxy. To learn more, refer to [Backend developer resources](https://grafana.com/docs/grafana/latest/developers/plugins/backend/#resources).
+- Use a custom data source request proxy. To learn more, refer to [Backend developer resources](./introduction/backend-plugins#resources).
```
Comment (sympatheticmoose) [inline docusaurus/docs/get-started/get-started.mdx:183]:
```suggestion
- Use a custom data source request proxy. To learn more, refer to [Backend developer resources](../introduction/backend-plugins#resources).
```

The website seems to figure this out, but I think should be `..`

File: docusaurus/docs/introduction/backend.md [hunk 4]
```diff
@@ -57,7 +57,7 @@ Grafana's backend plugin system exposes several key capabilities, or building bl
 
 #### Query data
 
-The query data capability allows a backend plugin to handle data source queries that are submitted from a [dashboard](https://grafana.com/docs/grafana/latest/dashboards), [Explore](https://grafana.com/docs/grafana/latest/explore) or [Grafana Alerting](https://grafana.com/docs/grafana/latest/alerting). The response contains [data frames](https://grafana.com/docs/grafana/latest/developers/plugins/introduction-to-plugin-development/data-frames/), which are used to visualize metrics, logs, and traces.
+The query data capability allows a backend plugin to handle data source queries that are submitted from a [dashboard](https://grafana.com/docs/grafana/latest/dashboards), [Explore](https://grafana.com/docs/grafana/latest/explore) or [Grafana Alerting](https://grafana.com/docs/grafana/latest/alerting). The response contains [data frames](../introduction/data-frames.md), which are used to visualize metrics, logs, and traces.
```
Comment (sympatheticmoose) [inline docusaurus/docs/introduction/backend.md:60]:
```suggestion
The query data capability allows a backend plugin to handle data source queries that are submitted from a [dashboard](https://grafana.com/docs/grafana/latest/dashboards), [Explore](https://grafana.com/docs/grafana/latest/explore) or [Grafana Alerting](https://grafana.com/docs/grafana/latest/alerting). The response contains [data frames](./data-frames.md), which are used to visualize metrics, logs, and traces.
```

As with the above, doesn't seem to break, but seems odd.

**Review by sympatheticmoose — CHANGES_REQUESTED** [review]

File: docusaurus/docs/get-started/get-started.mdx (hunk 3 above)
Comment (josmperez) [reply docusaurus/docs/get-started/get-started.mdx:183]: For some reason, your suggest produces a "Page Not Found" error, but the original link works. I'm not sure why, but @jackw may know.

Comment (jackw) [reply docusaurus/docs/get-started/get-started.mdx:183]: It's due to the `slug: /` in the metadata at the top of the file. This means where it lives on the file system (e.g. inside `get-started` directory) is irrelevant. Check the other internal links in this file and you'll see none of them try to go up a level out of the current directory. As such I don't think this suggestion is gonna work.

File: docusaurus/docs/ui-extensions/register-an-extension.md [hunk 5]
```diff
@@ -18,7 +18,7 @@ keywords:
 Be sure your plugin meets the following requirements before proceeding:
 
 - It must be an app plugin.
-- It must be preloaded by setting the [preload property](https://grafana.com/docs/grafana/latest/developers/plugins/metadata/#pluginjson) to `true` in the `plugin.json`.
+- It must be preloaded by setting the [preload property](https://grafana.com/developers/plugin-tools/reference-plugin-json#properties) to `true` in the `plugin.json`.
```
Comment (jackw) [inline docusaurus/docs/ui-extensions/register-an-extension.md:21]: nit: I think this should be an internal link?

**Review by jackw — APPROVED** [review]
LGTM.

Left a nit about a production link to plugin.json file.

**Review by sympatheticmoose — APPROVED** [review]
Approved - agree with Jacks suggestion on internal link

**Pull request merged**